src/formatDate.js

```diff
@@ -9,6 +9,59 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const separatorOld = fromFormat[fromFormat.length - 1];
+  const separatorNew = toFormat[toFormat.length - 1];
+
+  const fromFormatSlice = fromFormat.slice(0, 3);
+  const dateSplit = date.split(separatorOld);
+  const result = [];
+
+  const dateObject = {};
+
+  for (const type in fromFormatSlice) {
+    if (fromFormatSlice[type] === 'YYYY') {
+      dateObject.year = dateSplit[type];
+    }
+
+    if (fromFormatSlice[type] === 'YY') {
+      dateObject.year = dateSplit[type];
+    }
+
+    if (fromFormatSlice[type] === 'MM') {
+      dateObject.month = dateSplit[type];
+    }
+
+    if (fromFormatSlice[type] === 'DD') {
+      dateObject.day = dateSplit[type];
+    }
+  }
+
+  if (dateObject.year && dateObject.year.length === 2) {
+    const yearNum = parseInt(dateObject.year, 10);
+
+    dateObject.year =
+      yearNum < 30 ? `20${dateObject.year}` : `19${dateObject.year}`;
```

Comment:
The logic for handling two-digit years is correct, but it could be made more explicit by using a separate variable to store the century prefix and then concatenating it with the two-digit year.